### PR TITLE
check GDK_SCALE for screenshots

### PR DESCRIPTION
**Pull request by Noiseko**
fix for https://github.com/runelite/runelite/issues/11029

### CODE REVIEW

**Comment by Adam-** [comment]
1) is #11029 truly *only* with gpu enabled?
2) this will crash on any system where GDK_SCALE isn't defined

**Comment by Noiseko** [comment]
1. Yes this is completely only when GPU is enabled.
2. Ah, I guess you have a solution of that then?

**Comment by Adam-** [comment]
Just add a nullcheck to `getenv`?

**Comment by Noiseko** [comment]
okay think it's addressed now?

File: runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java [hunk 1]
```diff
@@ -1300,6 +1300,24 @@ private Image screenshot()
...
+				int gdkScale = Integer.parseInt(gdkScaleEnv);
+				if (gdkScale > 1)
+				{
+					width = width * gdkScale;
+					height = height * gdkScale;
+				}
+			}
+			catch (NumberFormatException e)
+			{
+				log.info("GDK_SCALE is not valid");
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java:1317]: this should probably be warn or debug. Also have you tested this with stretched mode?

Comment (Noiseko) [reply runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java:1317]: Yeah it works fine with stretched mode

File: runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java [hunk 2]
```diff
@@ -1300,6 +1300,24 @@ private Image screenshot()
...
+				int gdkScale = Integer.parseInt(gdkScaleEnv);
+				if (gdkScale > 1)
+				{
+					width = width * gdkScale;
+					height = height * gdkScale;
+				}
+			}
+			catch (NumberFormatException e)
+			{
+				log.warn("GDK_SCALE is not valid");
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/gpu/GpuPlugin.java:1317]: Log `gdkScaleEnv` so it's clear what it was actually set to

**Comment by Adam-** [comment]
Does it also work if you use the same logic the glViewport code uses?

```
			final AffineTransform t = ((Graphics2D) canvas.getGraphics()).getTransform();
			gl.glViewport(
				getScaledValue(t.getScaleX(), x),
				getScaledValue(t.getScaleY(), y),
				getScaledValue(t.getScaleX(), width),
				getScaledValue(t.getScaleY(), height));
```

**Comment by Nightfirecat** [comment]
Adam is asking if replacing the env var fetching with his code works the same way.

**Comment by Adam-** [comment]
> 
> 
> I don't know what you are suggesting. I appreciate my implementation might not be the best way to fix this, I'll happily test a more suitable fix?

eg. Use `getScaledValue(t.getScaleX(), width)` / `getScaledValue(t.getScaleY(), height)` as the width/height instead.

**Comment by Adam-** [comment]
fixed in #12594

**Pull request closed without merging**